docs/data-masking-overview.md

```diff
@@ -14,16 +14,43 @@ Data masking in Percona Server for MySQL is an essential tool for protecting sen
 
 These examples underscore how data masking serves as a crucial safeguard for sensitive information, allowing organizations to leverage their data effectively across diverse functions.
 
-Data masking helps to limit the exposure of sensitive data by preventing access to non-authorized users. Masking provides a way to create a version of the data in situations, such as a presentation, sales demo, or software testing, when the real data should not be used. Data masking changes the data values while using the same format and cannot be reverse engineered. Masking reduces an organization's risk by making the data useless to an outside party.
+
+## Version updates
+
+Percona Server for MySQL 8.0.41 introduces performance improvements for data masking through an internal term cache. The cache affects the following functions in the [data masking component](data-masking-function-list.md):
+
+* [gen_blocklist()](data-masking-function-list.md#gen_blockliststr-from_dictionary_name-to_dictionary_name)
+
+* [gen_dictionary()](data-masking-function-list.md#gen_dictionarydictionary_name)
+
+The new cache stores dictionary data in memory, making lookups faster than the previous method of querying the `mysql.masking_dictionaries` table each time. This speed boost is especially noticeable when you're working with many rows of data.
+
+However, the cache brings some new considerations. If you change the dictionary table directly (instead of using the proper dictionary management functions), your cache and table data can become different. 
+
+The dictionary manipulation functions are the following:
+
+* [`masking_dictionary_term_add()`](data-masking-function-list.md#masking_dictionary_term_adddictionary_name-term_name)
+
+* [`masking_dictionary_term_remove()`](data-masking-function-list.md#masking_dictionary_term_removedictionary_name-term_name)
+
+* [`masking_dictionary_remove()`](data-masking-function-list.md#masking_dictionary_removedictionary_name)
+
+To fix this, you can use the new [`masking_dictionaries_flush()`](data-masking-function-list.md#masking_dictionaries_flush) function to sync them back up. This function returns `1` when successful.
+
+The changes also affect how row-based replication works. When dictionary changes happen on the source server, they travel through the binary log to the replica server. While the replica applies these changes to its table correctly, the dictionary term cache doesn't update right away.
+
+To handle this, there's a new system setting called  [`component_masking_functions.dictionaries_flush_interval_seconds()`](data-masking-function-list.html#dictionaries_flush_interval_secondsinteger-unsigned)
```

Comment:
```suggestion
To handle this, there's a new component system variable called  [`masking_functions.dictionaries_flush_interval_seconds()`](data-masking-function-list.html#dictionaries_flush_interval_secondsinteger-unsigned)
```

Reply:
@patrickbirch One more change we did in 8.0.41.
All dictionary manipulations functions:
* `masking_dictionary_remove()`
* `masking_dictionary_term_remove()`
* `masking_dictionary_term_add()`
now return INTEGER instead of STRING.
Old behavior: string literal "1" - success, NULL - failure
New behavior: integer 1 - success, integer 0 - failure
Please, update the description of these functions as well (probably, no versioning comments needed, just make sure that we document the `new behavior`).

Reply:
@patrickbirch in one of the recent post-fixes we also had to limit the max value of the `dictionaries_flush_interval_seconds` variable to 31 536 000 (60 * 60 * 24 * 365 seconds = 1 year). This needs to be reflected in the documentation as well.